synapse/pipeline.yml

```diff
@@ -33,7 +33,7 @@ steps:
   - label: ":newspaper: Newsfile"
     command:
       - "python -m pip install tox"
-      - "scripts-dev/check-newsfragment"
+      - "scripts-dev/check-newsfragment || echo 'Please see the contributing guide for help: https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#changelog'"
```

Comment:
I suspect this will exit with a zero exitcode on failure, so the buildstep won't reject invalid newsfragments.

tbh I think you were probably right the first time: it belongs in the script, not here.

Reply:
The script has `set -e`, so any non-zero exit on a command will exit the script with a nonzero exit code. And the script specifically specifies `exit 1` for any errors it encounters.

Putting it in the script would require some refactoring of the script to cover all error cases.

Reply:
> Putting it in the script would require some refactoring of the script to cover all error cases.

Maybe? It's not actually obvious to me we want to print this out every time the script exits non-zero.

The script already has user-friendly error messages for the cases where it's the user's fault rather than something else. I suggest it makes more sense to improve those errors than it does to add *more* text here. 

If the goal is to avoid repeating the "Please see the guide..." text, just stick that in a variable in the script so that it can be referenced in each error message.

Reply:
I see what you mean. We indeed wouldn't want `Please see the contributing guide...` to appear if `git fetch -q origin develop` fails.

I think that a link to the contributing guide being included on the tail end of the each of the errors in the script does make sense though, especially if the user has made multiple errors. They should read the guide and fix all of them instead of going through each via trial-and-error.

I'll try to modify the script instead, including detected `tox`'s error code inside the script so the link text output is limited to that failing, rather than other random bits of the script. Thanks for bearing with me on this :slightly_smiling_face: